internal/gatewayapi/envoypatchpolicy.go

```diff
@@ -30,21 +40,35 @@ func (t *Translator) ProcessEnvoyPatchPolicies(envoyPatchPolicies []*egv1a1.Envo
 			irKey       string
 		)
 
+		refKind, refName := policy.Spec.TargetRef.Kind, policy.Spec.TargetRef.Name
 		if t.MergeGateways {
 			targetKind = resource.KindGatewayClass
 			irKey = string(t.GatewayClassName)
 
 			ancestorRef = gwapiv1a2.ParentReference{
 				Group: GroupPtr(gwapiv1.GroupName),
 				Kind:  KindPtr(targetKind),
-				Name:  policy.Spec.TargetRef.Name,
+				Name:  refName,
+			}
+
+			if string(refKind) == targetKind && refName != t.GatewayClassName {
+				// If the TargetRef name is not the same as the GatewayClass name, then skip processing.
+				// TODO: report status?
+				continue
```

Comment:
we have similar logic on L79
```
		gwXdsIR, ok := xdsIR[irKey]
		if !ok {
			continue
		}
```

can we modify it to fix this issue ? it seems to be we are incorrectly generating irKey

Reply:
good catch, we can remove the gwSet check.
but L79 won't work for gatewaclass case.

Reply:
yes this is the case we are trying to solve
when we are processing GWC foo, with Policy targeting GWC bar, we need to skip / continue. The point I was making was reuse the IR key logic here which is also valid for GWC https://github.com/envoyproxy/gateway/blob/b0b1b71d61657a8984ce53b590531ed7a2258ba2/internal/gatewayapi/translator.go#L432

Reply:
also impacts this line

```
		// Append the IR
		gwXdsIR.EnvoyPatchPolicies = append(gwXdsIR.EnvoyPatchPolicies, &policyIR)
```